ms2pip/spectrum_output.py

```diff
@@ -653,9 +654,9 @@ def open(self):
     def write(self, processing_results: List[ProcessingResult]):
         """Write MS2PIP predictions to a DLIB SQLite file."""
         connection = self._file_object
-        dlib.metadata.create_all()
+        dlib.metadata.create_all(connection.engine)
         self._write_metadata(connection)
-        self._write_entries(processing_results, connection, self.filename)
+        self._write_entries(processing_results, connection, str(self.filename))
```

Comment:
Is `self.filename` a Path?

Reply:
Yup, but it was already being converted into a string downstream in the `_write_entries` method. The type hint was simply too stringent, which led me to make a change here. I now reverted the addition here and updated the function signature type hint. I also properly used `os.PathLike` throughout the module for type hints, instead of `pathlib.Path`.